katdal/dataset.py

```diff
@@ -799,6 +800,16 @@ def select(self, **kwargs):
                         continue
                     target_keep |= (target_index_sensor == target_index)
                 self._time_keep &= target_keep
+            elif k == 'target_tags':
+                tags = v if is_iterable(v) else [v]
+                target_keep = np.zeros(len(self._time_keep), dtype=np.bool)
+                target_index_sensor = self.sensor.get('Observation/target_index')
+                for target in self.catalogue.targets:
+                    target_tags = target.tags
+                    if set(target_tags) & set(tags):
+                        target_index = self.catalogue.targets.index(target)
+                        target_keep |= (target_index_sensor == target_index)
+                self._time_keep &= target_keep
```

Comment:
It's worth collecting all the tags in the catalogue in a set and highlighting any unknown strings in `tags` at the end to help the user debug the selection, using `logger.warning` like in the `targets` selection.

Reply:
Done.